llpc/translator/lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -933,7 +933,9 @@ class SPIRVBranchConditional : public SPIRVInstruction {
     assert(getTrueLabel()->isForward() || getTrueLabel()->isLabel());
     assert(getFalseLabel()->isForward() || getFalseLabel()->isLabel());
 #if SPV_VERSION >= 0x10600
-    assert(TrueLabelId != FalseLabelId);
+    // This requirement was added in 1.6, but we also need to accept SPIR-V
+    // from before that, so ignore violations.
+    // assert(TrueLabelId != FalseLabelId);
```

Comment:
I guess we should change it to this, similar to what we have done for new GFXIP:
```
#if SPV_VERSION >= 0x10600
  if (spv::Version >= 0x10600)
    assert(TrueLabelId != FalseLabelId);
#endif
```

Reply:
I don’t think that works. spv::Version is a constant, so the if condition would always evaluate to true, failing the assert if we’re receiving old SPIR-V.
We would need `if (header->spvVersion >= 0x10600)` but I think we have no access to the SPIR-V header inside the validate function.

Reply:
Oh, sorry, you are right. I mistakenly express my intention. Yes, in this case, we actually need to check the SPIR-V binary version dynamically. I am fine with commenting it off as a temp solution.